### PR TITLE
Added missing documentation sections to make_saveable_from_iterator, description corrected.

### PR DESCRIPTION
**Pull request by Kab1r**
Added returns and raises section to make_saveable_from_iterator function #29406

@rthadur 
Pull request is now against master.
Recommitted for easier merging into master.

### CODE REVIEW

**Review by alextp — DISMISSED** [review]

**Review by jsimsa — CHANGES_REQUESTED** [review]
Remove trailing white space from this CL and update the name of this PR to reflect the change.

File: tensorflow/python/data/experimental/ops/iterator_ops.py [hunk 1]
```diff
@@ -33,6 +33,12 @@ def make_saveable_from_iterator(iterator):
 
   Args:
     iterator: Iterator.
+    
+  Returns:
+    A SaveableObject for saving/restore iterator state using Saver.
```
Comment (jsimsa) [inline tensorflow/python/data/experimental/ops/iterator_ops.py]: s/restore/restoring

File: tensorflow/python/data/experimental/ops/iterator_ops.py [hunk 2]
```diff
@@ -33,6 +33,12 @@ def make_saveable_from_iterator(iterator):
...
+	    
+  Raises:
+    ValueError: If iterators does not support checkpointing.
```
Comment (jsimsa) [inline tensorflow/python/data/experimental/ops/iterator_ops.py]: s/iterators/iterator

**Comment by KabirKwatra** [comment]
requested changes were made.

**Review by jsimsa — CHANGES_REQUESTED** [review]
@KabirKwatra thank you for the update. However, the PR description still incorrectly references `tf.image.convert_image_dtype`. Please update it to reference the method for which documentation is updated.

**Comment by KabirKwatra** [comment]
@jsimsa Sorry about the title, It has been corrected.

**Review by jsimsa — APPROVED** [review]

**Pull request merged**